subsys/usb/usb_c/usbc_tc_common.c

```diff
@@ -127,7 +127,9 @@ void tc_set_state(const struct device *dev, const enum tc_state_t state)
 	struct usbc_port_data *data = dev->data;
 	struct tc_sm_t *tc = data->tc;
 
+#if !defined(CONFIG_TEST)
```

Comment:
Why remove the assert when `CONFIG_TEST` is enabled? The point behind asserts is to alert programmer about conditions that must never be false but do become false due to incorrect code (e.g. invalid API usage).

Reply:
I agree that this is not the correct thing to do but twister was complaining:
```
/home/sam/Projects/github/zephyrproject/zephyr/include/zephyr/sys/util.h:109:45: error: invalid application of 'sizeof' to incomplete type 'const struct smf_state[]'
  109 |         ((size_t) (IS_ARRAY(array) + (sizeof(array) / sizeof((array)[0]))))
      |                                             ^
/home/sam/Projects/github/zephyrproject/zephyr/include/zephyr/sys/__assert.h:116:23: note: in definition of macro '__ASSERT'
  116 |                 if (!(test)) {                                            \
      |                       ^~~~
/home/sam/Projects/github/zephyrproject/zephyr/subsys/usb/usb_c/usbc_tc_common.c:130:26: note: in expansion of macro 'ARRAY_SIZE'
  130 |         __ASSERT(state < ARRAY_SIZE(tc_states), "invalid tc_state %d", state);
```
But the build works fine.